axlearn/common/flash_attention/gpu_attention_test.py

```diff
@@ -346,6 +357,9 @@ def test_cudnn_against_triton_ref(
     causal: bool,
     dtype: jnp.dtype,
 ):
+    if jax.default_backend() == "cpu":
```

Comment:
Likewise, let's avoid assuming that the backend is either gpu or cpu in multiple places.

```suggestion
    if jax.default_backend() != "gpu":
```

Reply:
I'll leave this code as-is as you asked
> Nit: can we check it against "cpu" directly instead of != "gpu"?

In addition, at the begin of file, it allows only "gpu" and "cpu". So `== "cpu"` is `!= "gpu"` in this code.
```
if jax.default_backend() not in ("gpu", "cpu"):
```

Reply:
> In addition, at the begin of file, it allows only "gpu" and "cpu". So == "cpu" is != "gpu" in this code.

I know you are making this assumption, but such dependency is fragile---what if we extend the supported backends in the future?

In this case, requiring the backend to be "gpu" is both more robust and readable. What's the downside?